Rebased onto `main` and applied the `Rank-up moves` item from the latest review.

**"Confirm focused sanitizer validation on the exact merge or rebased head."** — done. Head is now `c56395f79`, rebased onto `73c45824e` (branch was 17 commits behind). The rebase is a mechanical refresh; the diff is unchanged by it — production +1 net (2 added, 1 removed), tests +18.

The regression test still fails on pre-fix source at this base (production change reverted, new tests kept):

```
 × preserves angle-addr email addresses
AssertionError: expected 'Contact us at Support  or reply here'
              to be 'Contact us at Support <support@example.com> or reply here'
 Tests  1 failed | 48 passed (49)
```

Focused and sibling outbound suites on the rebased head, all green:

```
src/infra/outbound/sanitize-text.test.ts                 49 passed (49)
src/infra/outbound/markdown-details.test.ts              14 passed (14)
src/infra/outbound/payloads.test.ts                      43 passed (43)
src/infra/outbound/deliver-payload.presentation.test.ts   3 passed (3)
src/infra/outbound/format.test.ts                         9 passed (9)
```

On `mantis: telegram-visible-proof`: the Mantis capture command is maintainer-triggered, so I have not posted it. If the terminal proof above is not enough for the changed path, I can add a mock-gateway harness run instead.

--- a/extensions/imessage/src/monitor/sanitize-outbound.test-support.ts
+++ b/extensions/imessage/src/monitor/sanitize-outbound.test-support.ts
@@ -1,4 +1,5 @@
 // Imessage test support covers sanitize outbound plugin behavior.
+import { sanitizeForPlainText } from "openclaw/plugin-sdk/channel-outbound";
 import { describe, expect, it } from "vitest";
 import {
   protectIMessageFencedRoleMarkers,
@@ -95,6 +96,14 @@ describe("sanitizeOutboundText", () => {
     ]) {
       expect(sanitizeIMessageFinalOutboundText(source).text).toBe(source);
     }
+  });
+
+  it("strips colon-qualified tags before the final security projection", () => {
+    const text = sanitizeForPlainText("<vendor:note>visible</vendor:note>", {
+      style: "markdown",
+    });
+
+    expect(sanitizeIMessageFinalOutboundText(text).text).toBe("visible");
   });
 
   it.each([

--- a/extensions/imessage/src/monitor/sanitize-outbound.ts
+++ b/extensions/imessage/src/monitor/sanitize-outbound.ts
@@ -25,7 +25,7 @@ const PRIVATE_USE_START = 0xe000;
 const PRIVATE_USE_END = 0xf8ff;
 const MAX_PRIVATE_MARKDOWN_UNWRAP_PASSES = 32;
 // Mirror the private shared plain-text HTML grammar to inspect every destructive removal stage.
-const PLAIN_TEXT_HTML_TAG_RE = /<\/?[a-z][a-z0-9_-]*\b[^>]*>/gi;
+const PLAIN_TEXT_HTML_TAG_RE = /<\/?[a-z][a-z0-9_-]*(?=[\s/>])[^>]*>/gi;
 const PRIVATE_PROVIDER_TAG_NAMES = [
   "think",
   "thinking",

--- a/extensions/imessage/src/monitor/sanitize-outbound.ts
+++ b/extensions/imessage/src/monitor/sanitize-outbound.ts
@@ -25,7 +25,7 @@ const PRIVATE_USE_START = 0xe000;
 const PRIVATE_USE_END = 0xf8ff;
 const MAX_PRIVATE_MARKDOWN_UNWRAP_PASSES = 32;
 // Mirror the private shared plain-text HTML grammar to inspect every destructive removal stage.
-const PLAIN_TEXT_HTML_TAG_RE = /<\/?[a-z][a-z0-9_-]*(?=[\s/>])[^>]*>/gi;
+const PLAIN_TEXT_HTML_TAG_RE = /<\/?[a-z][a-z0-9_.:-]*(?=[\s/>])[^>]*>/gi;
 const PRIVATE_PROVIDER_TAG_NAMES = [
   "think",
   "thinking",

--- a/src/infra/outbound/sanitize-text.test.ts
+++ b/src/infra/outbound/sanitize-text.test.ts
@@ -82,6 +82,12 @@ describe("sanitizeForPlainText", () => {
     expect(sanitizeForPlainText('<a href="https://example.com">link</a>')).toBe("link");
   });
 
+  it("strips colon- and dot-qualified tags", () => {
+    expect(
+      sanitizeForPlainText("<vendor:note>one</vendor:note><vendor.note>two</vendor.note>"),
+    ).toBe("onetwo");
+  });
+
   it("keeps stripping tags exposed by malformed tag text", () => {
     const sanitized = sanitizeForPlainText(
       "before <<script>script>alert(1)</<script>script> after",

--- a/src/infra/outbound/sanitize-text.test.ts
+++ b/src/infra/outbound/sanitize-text.test.ts
@@ -174,17 +174,9 @@ describe("sanitizeForPlainText", () => {
     expect(sanitizeForPlainText("Contact us at Support <support@example.com> or reply here")).toBe(
       "Contact us at Support <support@example.com> or reply here",
     );
-    expect(sanitizeForPlainText("Ping <alice@example.org> and <bob@example.net>")).toBe(
-      "Ping <alice@example.org> and <bob@example.net>",
-    );
   });
 
   it("still strips tags whose name ends at a tag boundary", () => {
-    // The angle-addr carve-out keys on the character after the tag name, so
-    // real markup, mention aliases, and scaffolding must keep stripping.
-    expect(sanitizeForPlainText('Mail <a href="mailto:support@example.com">us</a> now')).toBe(
-      "Mail us now",
-    );
     expect(sanitizeForPlainText("Ping <users/abc> for access")).toBe("Ping  for access");
   });
 

--- a/src/infra/outbound/sanitize-text.ts
+++ b/src/infra/outbound/sanitize-text.ts
@@ -8,7 +8,7 @@ import { stripInternalRuntimeScaffolding } from "./protocol-scaffolding.js";
 export { stripInternalRuntimeScaffolding };
 
 // A tag name ends at whitespace, `/`, or `>`; `<user@example.com>` is prose, not markup.
-const HTML_TAG_RE = /<\/?[a-z][a-z0-9_-]*(?=[\s/>])[^>]*>/gi;
+const HTML_TAG_RE = /<\/?[a-z][a-z0-9_.:-]*(?=[\s/>])[^>]*>/gi;
 const MAY_CONTAIN_MARKDOWN_CODE_RE = /[`~]|\t| {4}/;
 const CODE_ESCAPE = "\u0000e";
 const CODE_PLACEHOLDER = "\u0000p";

